tests/test_attribute_definition.py

```diff
@@ -31,14 +31,13 @@ class BoolData(Data[bool]):
     name = "bool"
 
     @staticmethod
-    def parse_parameter(parser: Parser) -> bool:
-        val = parser.parse_optional_ident()
-        if val == "True":
+    def parse_parameter(parser: BaseParser) -> bool:
+        val = parser.tokenizer.next_token_of_pattern('(True|False)')
```

Comment:
I would suggest having code like
```
if parser.parse_optional_string("True") is not None:
    ...
parser.parse_string("False")
...
```
Which is what MLIR people usually write, and to me makes more sense in that case (since we only have 2 cases).
However, we should definitely keep the regex for other use cases.